ros/src/tl_detector/tl_detector.py

```diff
@@ -128,9 +133,25 @@ def get_light_state(self, light):
             self.prev_light_loc = None
             return False
         cv_image = self.bridge.imgmsg_to_cv2(self.camera_image, "bgr8")
-        return self.light_classifier.get_classification(cv_image)
+        if not SAVING_IMAGES:
+            traffic_color = self.light_classifier.get_classification(cv_image)
+            return traffic_color
+        else:
+            self.image_counter += 1
+            save_file = "../../../images/{}-{}.jpeg".format(self.traffic_color_to_file_name(light.state),
```

Comment:
Duplication of config setting for training data folder. As mentioned, we can make this a ros param and read it at the start.